src/Microsoft.AspNetCore.Mvc.Razor/Compilation/ViewsFeatureProvider.cs

```diff
@@ -7,98 +7,58 @@
 using System.Linq;
 using System.Reflection;
 using Microsoft.AspNetCore.Mvc.ApplicationParts;
-using Microsoft.AspNetCore.Razor.Hosting;
+using Microsoft.AspNetCore.Mvc.Razor.Internal;
+using Microsoft.Extensions.Primitives;
 
 namespace Microsoft.AspNetCore.Mvc.Razor.Compilation
 {
     /// <summary>
     /// An <see cref="IApplicationFeatureProvider{TFeature}"/> for <see cref="ViewsFeature"/>.
     /// </summary>
+    [Obsolete("This type is obsolete and will be removed in a future version. See " + nameof(IRazorCompiledItemProvider) + " for alternatives.")]
     public class ViewsFeatureProvider : IApplicationFeatureProvider<ViewsFeature>
```

Comment:
The plan is for this type to only support MvcPrecompilation. When we stop supporting it, we can remove the type. 

Here's what's happening in this diff:

* Reset the code back to 2.0
* Added an Obsolete attribute. We want people to use the new interface instead of extending this type
* Moved code that was checked for duplicates from RazorViewCompiler that detected duplicates to this type. It's actually more appropriate since the exception was for files with different casings coming from a single

Reply:
How does the old feature interact with the new way of doing things. Specifically packages with existing precompiled views using the old way/format.
* Can those views be overriden?
* Do they have the right precedence compared to other views?

Reply:
Yup, we collate all ways when we build the cache entry i.e. after all feature providers have executed. We also register the `ViewsFeatureProvider`  to run after `RazorCompiledItemFeatureProvider`. Consequently, the precedence ends up being:

1. Views from App's Views.dll
2. View from Library Views.dll
3. Views from App's PrecompiledViews.dll <sup>+</sup>
4. Views from Library's PrecompiledViews.dll

<sup>+</sup>: I spoke to @DamianEdwards about this - he was ok requiring using the Razor Sdk to build your views if you'd like your app's views to override library views.